src/lib/gadgets/gadgets.ts

```diff
@@ -139,4 +143,36 @@ const Gadgets = {
   and(a: Field, b: Field, length: number) {
     return and(a, b, length);
   },
+
+  /**
+   * Multi-range check.
+   *
+   * Proves that x, y, z are all in the range [0, 2^88).
+   *
+   * This takes 4 rows, so it checks 88*3/4 = 66 bits per row. This is slightly more efficient
+   * than 64-bit range checks, which can do 64 bits in 1 row.
+   *
+   * In particular, the 3x88-bit range check supports bigints up to 264 bits, which in turn is enough
+   * to support foreign field multiplication with moduli up to 2^259.
+   */
+  multiRangeCheck(x: Field, y: Field, z: Field) {
```

Comment:
It would be great to match the template similar to the other gates, such as having an `@example` block, describing inputs, if it can throw, etc, just for consistency stake.